internal/services/formatters/go/nancy/formatter_test.go

```diff
@@ -16,145 +16,198 @@ package nancy
-
 		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return(output, nil)
 
-		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, config)
+		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, cfg)
 
 		formatter := NewFormatter(service)
-
-		assert.NotPanics(t, func() {
-			formatter.StartAnalysis("")
-		})
+		formatter.StartAnalysis("")
 
 		assert.Len(t, analysis.AnalysisVulnerabilities, 2)
+
+		for _, vuln := range analysis.AnalysisVulnerabilities {
+			assert.Equal(t, languages.Go, vuln.Vulnerability.Language, "Expected Go as vulnerability language")
+			assert.Equal(t, tools.Nancy, vuln.Vulnerability.SecurityTool, "Expected nancy as security tool")
+			assert.NotEmpty(t, vuln.Vulnerability.Severity, "Expected not empty vulnerability severity")
+			assert.NotEmpty(t, vuln.Vulnerability.Details, "Expected not empty vulnerability details")
+			assert.Equal(t, confidence.High, vuln.Vulnerability.Confidence, "Expected high as vulnerability confidence")
+			assert.NotEmpty(t, vuln.Vulnerability.Code, "Expected not empty vulnerability code")
+			assert.NotEmpty(t, vuln.Vulnerability.Line, "Expected not empty vulnerability line")
+			assert.Equal(
+				t,
+				filepath.Join(cfg.ProjectPath, "go.mod"),
+				vuln.Vulnerability.File,
+				"Expected equals vulnerability file",
+			)
+		}
 	})
 
 	t.Run("should success parse output empty to analysis", func(t *testing.T) {
-		analysis := &entitiesAnalysis.Analysis{}
+		analysis := new(analysis.Analysis)
 
-		config := &cliConfig.Config{}
-		config.WorkDir = &workdir.WorkDir{}
+		cfg := config.New()
 
 		dockerAPIControllerMock := testutil.NewDockerMock()
 		dockerAPIControllerMock.On("SetAnalysisID")
+		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return("", nil)
 
-		output := ""
-
-		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return(output, nil)
-
-		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, config)
+		service := formatters.NewFormatterService(analysis, dockerAPIControllerMock, cfg)
 
 		formatter := NewFormatter(service)
-
-		assert.NotPanics(t, func() {
-			formatter.StartAnalysis("")
-		})
+		formatter.StartAnalysis("")
 
 		assert.Len(t, analysis.AnalysisVulnerabilities, 0)
 	})
 
-	t.Run("should return error when parsing invalid output", func(t *testing.T) {
-		analysis := &entitiesAnalysis.Analysis{}
+	t.Run("should add error on analysis when parsing invalid output", func(t *testing.T) {
+		analysis := new(analysis.Analysis)
 
-		config := &cliConfig.Config{}
-		config.WorkDir = &workdir.WorkDir{}
+		cfg := config.New()
 
 		dockerAPIControllerMock := testutil.NewDockerMock()
 		dockerAPIControllerMock.On("SetAnalysisID")
-		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").
-			Return("invalid output", nil)
+		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return("invalid output", nil)
```

Comment:
in nancy analysis can return 2 types of the error:
1º Output wrong, how you create here,
2º Nancy return error when Github return this [nancy-output-error.txt](https://github.com/ZupIT/horusec/files/7595739/nancy-output-error.txt) i was this error can be describe in your tests and if possible create anyware treatment on formatter.

Reply:
But what should we do when we get this output from github? I think that the process will be the same for both outputs, or not?

Reply:
i think not, because we can create treatment for show correctly log for the user change in your configuration.
Currently when this error appers at the end of the analysis `Can'ot possible convert 'C' to struct...` but the error is because user not stay using `GITHUB_TOKEN` environment variable in analysis.

I think the correct is create validation for this case and apresent log correctly for the user change configuration and run it's analysis without errors.

Reply:
the log I added above is displayed only debug mode, other point to verify again this error, because generally any user run analysis in debug mode to see this error

Reply:
I agree. What do you think of making this another PR? I think it would be better to review and for the history

Reply:
humm, okay! I'll create issue about this too :)